src/tests/uat/enriched_events_uat.py

```diff
@@ -0,0 +1,275 @@
+# SPDX-License-Identifier: MIT
+# *******************************************************
+# *
+# * DISCLAIMER. THIS PROGRAM IS PROVIDED TO YOU "AS IS" WITHOUT
+# * WARRANTIES OR CONDITIONS OF ANY KIND, WHETHER ORAL OR WRITTEN,
+# * EXPRESS OR IMPLIED. THE AUTHOR SPECIFICALLY DISCLAIMS ANY IMPLIED
+# * WARRANTIES OR CONDITIONS OF MERCHANTABILITY, SATISFACTORY QUALITY,
+# * NON-INFRINGEMENT AND FITNESS FOR A PARTICULAR PURPOSE.
+
+"""
+The following API calls are tested in this script.
+
+For the validation CBC API requests are used.
+
+To execute, a profile must be provided using the standard CBC Credentials.
+
+Processes:
+- Get alert and get the enriched events for that alert
+"""
+
+# Standard library imports
+import datetime
+from pprint import pprint
+import requests
+import sys
+import time
+
+# Internal library imports
+from cbc_sdk.helpers import build_cli_parser, get_cb_cloud_object
+from cbc_sdk.endpoint_standard import EnrichedEvent, EnrichedEventFacet
+from cbc_sdk.platform.alerts import CBAnalyticsAlert
+from cbc_sdk.platform import Process
+
+
+HEADERS = {"X-Auth-Token": "", "Content-Type": "application/json"}
+ORG_KEY = ""
+HOSTNAME = ""
+
+# Formatters
+NEWLINES = 1
+DELIMITER = "-"
+SYMBOLS = 60
+TIMEOUT_MINS = 2
+
+# URLS
+START_JOB = "{}api/investigate/v2/orgs/{}/enriched_events/detail_jobs"
+STATUS_JOB = "{}api/investigate/v2/orgs/{}/enriched_events/detail_jobs/{}"
+GET_RESULTS = "{}api/investigate/v2/orgs/{}/enriched_events/detail_jobs/{}/results"
+
+
+def start_search_job(alert_id):
+    """Start a search job"""
+    url = START_JOB.format(HOSTNAME, ORG_KEY)
+    result = requests.post(url, json={"alert_id": alert_id}, headers=HEADERS)
+    return result.json().get("job_id")
+
+
+def get_status_search_job(job_id):
+    """Check the status of a search job"""
+    url = STATUS_JOB.format(HOSTNAME, ORG_KEY, job_id)
+    return requests.get(url, headers=HEADERS)
+
+
+def wait_till_job_ready(job_id):
+    """Checking status of the job, till ready"""
+    timeout_time = datetime.datetime.now() + datetime.timedelta(minutes=TIMEOUT_MINS)
+    while True:
+        result = get_status_search_job(job_id).json()
+        searchers_contacted = result.get("contacted", 0)
+        searchers_completed = result.get("completed", 0)
+        if searchers_completed == searchers_contacted:
+            break
+        if searchers_contacted == 0:
+            time.sleep(0.5)
+            continue
+        if searchers_completed < searchers_contacted:
+            if timeout_time < datetime.datetime.now():
+                break
+
+        time.sleep(0.5)
+
+
+def get_results_search_job(job_id):
+    """Get the results of a search job"""
+    url = GET_RESULTS.format(HOSTNAME, ORG_KEY, job_id)
+    result = requests.get(url, headers=HEADERS).json()
+    found_results = result.get("num_found", 0)
+    if found_results == 0:
+        return []
+    return result.get("results", [])
+
+
+# TODO rewrite the functons below
```

Comment:
The rest of the functions are just copy paste from src/tests/uat/process_search_calls.py, disregard for now.

Reply:
https://jira.carbonblack.local/browse/CBAPI-2603.